### ISSUE TITLE
Deserializing list of bytes back to list of CKKSVector

### ISSUE THREAD

**Issue by efriel94** [OpenMined/TenSEAL#428]
## Question
**Describe your question in ONE SENTENCE.**
I'm encrypting a list that holds a dataset to a list of CKKSVector. For each index in that list I am serializing to bytes before sending over to the server. I have a gRPC server listening and reading in the bytes via Protobuf. I am successfully reading the bytes in but having issues reconstructing the bytes on the server end back into a list of CKKSVector. The client sends a hash along with the data for the server to verify. The checksums are differing.

See code:

I have two datasets which are being encrypted which return ```list[CKKSVector]```

```python
def encrypt_data(self,data,name):
     t_start = time()
     enc_X = [ts.ckks_vector(self.secret_ctx, i.tolist()) for i in data]
     t_elapsed = time() - t_start
     print("Encryption of {} ---> numpy array of size [m={:d}, n={:d}] took {:.4f} seconds".format(name,data.shape[0], data.shape[1], t_elapsed))
     return enc_X

encrypted_X = client_lr.encrypt_data(X_train,"X train")                    # Encrypt X training data
encrypted_y = client_lr.encrypt_data(y_train,"y train")                     # Encrypt y training data

print(f"encrypted X sha256: {get_sha256(encrypted_X).hexdigest()}")
print(f"encrypted y sha256: {get_sha256(encrypted_y).hexdigest()}\n")

# Serializing into bytes
enc_X_bytes = [i.serialize() for i in encrypted_X]
enc_y_bytes = [i.serialize() for i in encrypted_y]

# Server side - gRPC
# sending the bytes as chunks since the dataset is too large
# looping over each chunk and converting it to CKKSVector

buffer = list()
context_fhe = ts.context_from("public_context_file.txt")
for chunk in request_iterator:
      if chunk.sha512 != 0:
           hash_data = chunk.sha512
     buffer.append(ts.ckks_vector_from(context_fhe,chunk.data))
     print(f"recieved hash: {hash_data}")
     print(f"sha256sum server side: {get_sha256(buffer).hexdigest()}")
```

To summarize if I encrypt and serialize a list of CKKSVector to bytes which returns ```list[bytes]```, how can go backwards to reconstruct a list of CKKSVector which returns ```list[CKKSVector]```

**Comment by efriel94** [comment]
Attempts tried:

```python
# Get hash of Encrypted list of CKKSVector
print(f"encrypted X sha256: {get_sha256(encrypted_X).hexdigest()}")
print(f"encrypted y sha256: {get_sha256(encrypted_y).hexdigest()}\n")

# Serializing list of CKKSVector into bytes
enc_X_bytes = [i.serialize() for i in encrypted_X]
enc_y_bytes = [i.serialize() for i in encrypted_y]

# Get hash of list of bytes
print(f"encrypted X sha256: {get_sha256(enc_X_bytes).hexdigest()}")
print(f"encrypted y sha256: {get_sha256(enc_y_bytes).hexdigest()}\n")

def reconstruct(enc_bytes):
    """Create TenSEAL vector from byte vector"""
    ckks_vector = ts.lazy_ckks_vector_from(enc_bytes)
    ckks_vector.link_context(context_fhe)
    return ckks_vector

def reconstruct_list(enc_bytes_list):
    """Create TenSEAL vector from byte vector"""
    enc = []
    for elem in enc_bytes_list:
        enc.append(reconstruct(elem))
    return enc

# option 1 - wrong
ckks_list = reconstruct_list(enc_X_bytes)
print(get_sha256(ckks_list).hexdigest())

# option 2 - wrong
ckks_list = [ts.ckks_vector_from(context_fhe,i) for i in enc_X_bytes]
print(get_sha256(ckks_list).hexdigest())
```

**Comment by efriel94** [comment]
So the computation works as expected in the encrypted domain using LR even when the original hash ```list[CKKSVector]``` and the computed hash server side ```list[CKKSVector]``` differ but produce the same results and approx. similar accuracy. It might highlight that the functions are not a complete inverse of each other.
